pkg/controller/provider/web/ocp/base.go

```diff
@@ -273,9 +273,20 @@ func (h Handler) NetworkAttachmentDefinitions(ctx *gin.Context, provider *api.Pr
 		return
 	}
 
+	requestedNS := ctx.Request.URL.Query().Get(NsParam)
 	// Determine which namespaces to query
 	var namespacesToQuery []string
-	if provider != nil && provider.IsRestrictedHost() {
+
+	if requestedNS != "" {
+		if provider != nil && provider.IsRestrictedHost() {
+			// Validate the requested namespace is in allowed list
+			if requestedNS != provider.GetNamespace() && requestedNS != core.NamespaceDefault {
+				return nil, liberr.New("Namespace '%s' not allowed for restricted provider. Allowed namespaces: ['%s', 'default']",
+					requestedNS, provider.GetNamespace())
+			}
+		}
+		namespacesToQuery = []string{requestedNS}
+	} else if provider != nil && provider.IsRestrictedHost() {
 		// For restricted host providers, query specific namespaces
```

Comment:
```suggestion
        providerNS = provider.GetNamespace()
        defaultNS = core.NamespaceDefault
        queryNS = ctx.Request.URL.Query().Get(NsParam)
        
        // check if query ns is allowed
        if (queryNS != "" && queryNS != providerNS && queryNS != defaultNS) { // error ... }
        
		// For restricted host providers, query allowed namespaces
		namespacesToQuery = []string{providerNS, defaultNS}
```

this logic is identical to the logic you added from lines 279 to 289, it makes the issue clear:
when using restricted provider we can only get items from the provider ns and default ns

so the logic say:
```
if restricted provider:
     if qury ns is defined and not allowed:
          error
     fetch from allowed ns
```

Reply:
@Hazanel hi, after our of line discussion I think I understand the issue better,
I think the issue is not in the list NADs method here, where we should get all available NADs from all available namespaces, IMHO the issue is in methods that use the list of all available NADs and search for specific NAD using it's name only, while they should look for NAD using it's name and namespace.

@coderabbitai, hi, can you verify this method is used in places that expect all the NADs from all available namespaces ?
can you look for places we search the list of all the NADs, but filter using only name, without taking into account the namespace, fetching NADs that may have the same name but wrong namespaces ?